{
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "Following code will produce an error when compiled with mingw32 from anaconda in windows:\r\n\r\ncyfile.pyx\r\n```python\r\nimport numpy as np\r\ncimport numpy as np\r\nimport ctypes\r\n\r\ndef get_arr():\r\n    cdef np.ndarray[long double, ndim=1] outp = np.zeros(10, dtype=ctypes.c_longdouble)\r\n    return outp\r\n```\r\nsetup.py:\r\n```python\r\ntry:\r\n\tfrom setuptools import setup\r\nexcept:\r\n\tfrom distutils.core import setup\r\nimport numpy as np\r\nfrom distutils.extension import Extension\r\nfrom Cython.Distutils import build_ext\r\n\r\nsetup(\r\n    name  = \"cyext\",\r\n    packages = [\"cyext\"],\r\n    cmdclass = {'build_ext': build_ext},\r\n    ext_modules = [Extension(\"cyext\", sources=[\"cyfile.pyx\"], include_dirs=[np.get_include()])]\r\n    )\r\n```\r\n```python\r\n>>> import cyext\r\n>>> cyext.get_arr()\r\n```\r\n```\r\nTraceback (most recent call last):\r\n  File \"<stdin>\", line 1, in <module>\r\n  File \"cyfile.pyx\", line 6, in cyext.get_arr\r\n    cdef np.ndarray[long double, ndim=1] outp = np.zeros(10, dtype=ctypes.c_longdouble)\r\nValueError: Buffer dtype mismatch, expected 'long double' but got 'double'\r\n```\r\n",
    "closed_at": null,
    "comment_data": [
        {
            "author_association": "MEMBER",
            "body": "You should probably use the correct NumPy dtype, not anything from ctypes.\n",
            "created_at": "2019-09-07T21:39:00Z",
            "html_url": "https://github.com/cython/cython/issues/3136#issuecomment-529148891",
            "id": 529148891,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/3136",
            "node_id": "MDEyOklzc3VlQ29tbWVudDUyOTE0ODg5MQ==",
            "updated_at": "2019-09-07T21:39:00Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/529148891",
            "user": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/491659?v=4",
                "events_url": "https://api.github.com/users/scoder/events{/privacy}",
                "followers_url": "https://api.github.com/users/scoder/followers",
                "following_url": "https://api.github.com/users/scoder/following{/other_user}",
                "gists_url": "https://api.github.com/users/scoder/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/scoder",
                "id": 491659,
                "login": "scoder",
                "node_id": "MDQ6VXNlcjQ5MTY1OQ==",
                "organizations_url": "https://api.github.com/users/scoder/orgs",
                "received_events_url": "https://api.github.com/users/scoder/received_events",
                "repos_url": "https://api.github.com/users/scoder/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/scoder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/scoder/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/scoder"
            }
        },
        {
            "author_association": "NONE",
            "body": "> You should probably use the correct NumPy dtype, not anything from ctypes.\r\n\r\nStill throws the same error:\r\n```python\r\nimport numpy as np\r\ncimport numpy as np\r\nimport ctypes\r\n\r\ndef get_arr():\r\n\tcdef np.ndarray[long double, ndim = 1] outp = np.zeros(10, dtype = np.longdouble)\r\n\treturn outp\r\n```\r\n```python\r\nimport cyext\r\ncyext.get_arr()\r\n```\r\n```\r\nTraceback (most recent call last):\r\n  File \"<stdin>\", line 1, in <module>\r\n  File \"cyfile.pyx\", line 6, in cyext.get_arr\r\n    cdef np.ndarray[long double, ndim = 1] outp = np.zeros(10, dtype = np.longdouble)\r\nValueError: Item size of buffer (8 bytes) does not match size of 'long double' (16 bytes)\r\n```\r\n\r\n(Same result from `np.longdouble_t` in the cdef side)",
            "created_at": "2019-09-08T07:19:15Z",
            "html_url": "https://github.com/cython/cython/issues/3136#issuecomment-529177192",
            "id": 529177192,
            "issue_url": "https://api.github.com/repos/cython/cython/issues/3136",
            "node_id": "MDEyOklzc3VlQ29tbWVudDUyOTE3NzE5Mg==",
            "updated_at": "2019-09-08T07:20:52Z",
            "url": "https://api.github.com/repos/cython/cython/issues/comments/529177192",
            "user": {
                "avatar_url": "https://avatars0.githubusercontent.com/u/11860098?v=4",
                "events_url": "https://api.github.com/users/david-cortes/events{/privacy}",
                "followers_url": "https://api.github.com/users/david-cortes/followers",
                "following_url": "https://api.github.com/users/david-cortes/following{/other_user}",
                "gists_url": "https://api.github.com/users/david-cortes/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/david-cortes",
                "id": 11860098,
                "login": "david-cortes",
                "node_id": "MDQ6VXNlcjExODYwMDk4",
                "organizations_url": "https://api.github.com/users/david-cortes/orgs",
                "received_events_url": "https://api.github.com/users/david-cortes/received_events",
                "repos_url": "https://api.github.com/users/david-cortes/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/david-cortes/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/david-cortes/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/david-cortes"
            }
        }
    ],
    "comments": 2,
    "comments_url": "https://api.github.com/repos/cython/cython/issues/3136/comments",
    "created_at": "2019-09-07T21:25:41Z",
    "event_data": [],
    "events_url": "https://api.github.com/repos/cython/cython/issues/3136/events",
    "html_url": "https://github.com/cython/cython/issues/3136",
    "id": 490681881,
    "labels": [],
    "labels_url": "https://api.github.com/repos/cython/cython/issues/3136/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "MDU6SXNzdWU0OTA2ODE4ODE=",
    "number": 3136,
    "repository_url": "https://api.github.com/repos/cython/cython",
    "state": "open",
    "title": "Cannot create NumPy long double arrays with mingw",
    "updated_at": "2019-09-08T07:20:52Z",
    "url": "https://api.github.com/repos/cython/cython/issues/3136",
    "user": {
        "avatar_url": "https://avatars0.githubusercontent.com/u/11860098?v=4",
        "events_url": "https://api.github.com/users/david-cortes/events{/privacy}",
        "followers_url": "https://api.github.com/users/david-cortes/followers",
        "following_url": "https://api.github.com/users/david-cortes/following{/other_user}",
        "gists_url": "https://api.github.com/users/david-cortes/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/david-cortes",
        "id": 11860098,
        "login": "david-cortes",
        "node_id": "MDQ6VXNlcjExODYwMDk4",
        "organizations_url": "https://api.github.com/users/david-cortes/orgs",
        "received_events_url": "https://api.github.com/users/david-cortes/received_events",
        "repos_url": "https://api.github.com/users/david-cortes/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/david-cortes/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/david-cortes/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/david-cortes"
    }
}